### PR TITLE
levvy.fi token lending

### PR DESCRIPTION
**Pull request by theapesociety**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Please enable "Allow edits by maintainers" while putting up the PR.
3. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Levvy.fi Tokens

##### Twitter Link:
https://twitter.com/levvyfinance

##### List of audit links if any:
https://tas.nyc3.cdn.digitaloceanspaces.com/Levvy%20v2%20Audit.pdf

##### Website Link:
https://levvy.fi/

##### Logo (High resolution, will be shown with rounded borders):
https://levvy.fi/_next/image?url=%2Fimages%2FLogo%2FLevvyLogo.png&w=3840&q=100

##### Current TVL:
137.90 k USD

##### Chain:
Cardano

##### Short Description (to be shown on DefiLlama):
Levvy for Tokens is a P2P Orderbook Fungible Lending Protocol on Cardano.  Offers fixed APYs and durations with variable LTVs set by users.

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Lending

##### methodology (what is being counted as tvl, how is tvl being calculated):
TVL is calculated by ADA locked in lending pools and the value of tokens that are locked as collateral.

##### Github org/user (Optional, if your code is open source, we can track activity):

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/levvy-fi-tokens exports TVL: 
        
 
 
    cardano-staking           1.66 M
    cardano-borrowed          35.30 k
    cardano                   138.07 k
    staking                   1.66 M
    borrowed                  35.30 k
    
    total                    138.07 k

**Comment by theapesociety** [comment]
Referencing this conversation on discord regarding merging with existing Levvy.fi protocol into tabs: https://discord.com/channels/823822164956151810/823822165538373696/1159912460292206725

**Comment by g1nt0ki** [comment]
hi @theapesociety can you update adapter to compute tvl on chain?

**Comment by g1nt0ki** [comment]
we have basic support for querying cardano: https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/helper/chain/cardano.js

**Comment by theapesociety** [comment]
> hi @theapesociety can you update adapter to compute tvl on chain?

Issue i'm running into with this is that the smart contract generates a new address for every single loan, so I'm unsure if we can dynamically figure out the address of each loan each time.

File: projects/levvy-fi-tokens/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "Counts ADA locked in token lending offer pools, and tokens locked as collateral in loans",
+  misrepresentedTokens: true,
+  cardano: {
+    tvl: async () => {
+      const data = await post(
+        "https://citizens.theapesociety.io/api/getLevvyData",
+        {}
+      );
+      return {
+        cardano: data.tokens.tvl,
```
Comment (g1nt0ki) [inline projects/levvy-fi-tokens/index.js:14]: what is included in the tvl here? does not match the vaule on your site: 
image
Also, is tvl: total supplied or total supplied - total borrowed?

Comment (theapesociety) [reply projects/levvy-fi-tokens/index.js:14]: One the site, that number doesn't currently include the collateral locked in active loans.  TVL in this commit equals total in pools + total value of collateral in active loans

**Comment by theapesociety** [comment]
@g1nt0ki just saw the merge, thanks for the help on this and setting up the view options!  would it be possible that the middle tab says Levvy for NFTs and the 3rd Tab says Levvy for Tokens.

Also, can the overall category be lending and the icon be below for the full combined protocol?
levvy

**Pull request merged**